azure/scope/cluster.go

```diff
@@ -247,10 +248,47 @@ func (s *ClusterScope) PublicIPSpecs() []azure.ResourceSpecGetter {
 func (s *ClusterScope) LBSpecs() []azure.ResourceSpecGetter {
 	var specs []azure.ResourceSpecGetter
 	if s.ControlPlaneEnabled() {
-		specs = []azure.ResourceSpecGetter{
-			&loadbalancers.LBSpec{
-				// API Server LB
-				Name:                 s.APIServerLB().Name,
+		frontendLB := &loadbalancers.LBSpec{
```

Comment:
Why are we changing the way we construct the cluster LB in this way? I don't see any functional difference (indeed I wouldn't want to) and I don't see any of these new variables being re-used by the conditional ILB construction below.

Reply:
You are right, here is no functional difference.

We want to have separate variable defining for the APIServer LB because the logic to "save" frontendIPs has been pulled to a `for` loop. 
It's been pulled out because the frontendIPs can have at most two IPs.
It seemed cleaner to me to have it as a separate local variable.

Reply:
I think there is more value in optimizing for a leaner diff.

1) We aren't making any functional changes to the existing LB constructor outcomes
2) We are making a significant functional change within this function scope

If possible, we want the diff to be a reflection of the above, making #2 in particular as conspicuous as possible, to predict for better reviews (easier to read) and easier historical analysis ("what exactly did we do with that change?").

If there are other improvements to be made as part of the process of encountering this code surface area, let's capture that with a short description in an issue for working on in the future based on prioritization.